lib/route.js

```diff
@@ -208,117 +201,132 @@ function buildRouting (options) {
       }
 
       if (prefixing === false) {
-      // run 'onRoute' hooks
+        // run 'onRoute' hooks
         for (const hook of this[kHooks].onRoute) {
-          try {
-            hook.call(this, opts)
-          } catch (error) {
-            done(error)
-            return
-          }
+          hook.call(this, opts)
         }
       }
 
+      const constraints = opts.constraints || {}
       const config = {
         ...opts.config,
         url,
         method: opts.method
       }
-      const constraints = opts.constraints || {}
-      if (opts.version) {
-        warning.emit('FSTDEP006')
-        constraints.version = opts.version
-      }
-
       const context = new Context(
         opts.schema,
         opts.handler.bind(this),
         this[kReply],
         this[kRequest],
         this[kContentTypeParser],
-        config,
-        opts.errorHandler || this[kErrorHandler],
-        opts.bodyLimit,
-        opts.logLevel,
-        opts.logSerializers,
-        opts.attachValidation,
-        this[kReplySerializerDefault],
-        opts.schemaErrorFormatter || this[kSchemaErrorFormatter]
+        config
       )
 
-      const headRouteExists = router.find('HEAD', path) != null
-
-      try {
-        router.on(opts.method, opts.url, { constraints }, routeHandler, context)
-      } catch (err) {
-        done(err)
-        return
+      if (opts.version) {
+        warning.emit('FSTDEP006')
+        constraints.version = opts.version
       }
 
-      const { exposeHeadRoute } = opts
-      const hasRouteExposeHeadRouteFlag = exposeHeadRoute != null
-      const shouldExposeHead = hasRouteExposeHeadRouteFlag ? exposeHeadRoute : globalExposeHeadRoutes
+      const headRouteExists = opts.method === 'HEAD' && router.find('HEAD', url, constraints) != null
```

Comment:
Not sure why this behavior changed. It's needed for this feature?

Reply:
When moving to sync registering of routes, yes. This due to the fact that before the sibling `HEAD` route was being added on the avvio's `after` hook, causing no issues as because of the async nature. For instance, wasn't interfering with the proper validations of original `GET` route, getting the validations errors in the right order